File: 예림/pgs/달리기 경주.cpp
#include <bits/stdc++.h>
using namespace std;

map<string, int> m;    // {players, rank}

vector<string> solution(vector<string> players, vector<string> callings) {

	for (int i = 0; i < players.size(); i++) {
		m[players[i]] = i;
	}

	for (int i = 0; i < callings.size(); i++) {
		int nowIdx = m[callings[i]];
		string now = callings[i];

		int prevIdx = nowIdx - 1;
		string prev = players[prevIdx];

		swap(players[nowIdx], players[prevIdx]);

		m[now] = prevIdx;
		m[prev] = nowIdx;
	}

	return players;
}

File: 예림/pgs/달리기 경주.java
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        Map<String, Integer> hm = new HashMap<>();  // 이름, 등수
        
        for(int i = 0; i < players.length; i++) {
            hm.put(players[i], i);
        }
        
        for(String name : callings) {
            int nowRank = hm.get(name);
            String prevPlayer = players[nowRank - 1];
            
            hm.put(name, nowRank - 1);
            hm.put(prevPlayer, nowRank);
            
            players[nowRank - 1] = name;
            players[nowRank] = prevPlayer;
        }
        
        return players;
    }
}